apps/site-app/src/containers/Sidebar/index.tsx

```diff
@@ -0,0 +1,128 @@
+import type { SidebarLink } from '@thecointech/shared/containers/PageSidebar/types';
+import { defineMessages, useIntl } from 'react-intl';
+import { isLocal } from '@thecointech/signers';
+
+import { AppRoutes } from '../App/Routes';
+import { useLocation } from 'react-router';
+
+const translations = defineMessages({
+  home: {
+    defaultMessage: 'Overview',
+    description: 'app.accounts.sidebar.home: Title for the Home entry in the menu'
+  },
+  transferin: {
+    defaultMessage: 'Top up balance',
+    description: 'app.accounts.sidebar.transferin: Title for the transfer entry in the menu'
+  },
+  makepayments: {
+    defaultMessage: 'Make payments',
+    description: 'app.accounts.sidebar.makepayments: Title for the Make payments entry in the menu'
+  },
+  settings: {
+    defaultMessage: 'Settings',
+    description: 'app.accounts.sidebar.settings: Title for the Settings entry in the menu'
+  },
+  contact: {
+    defaultMessage: 'Contact Us',
+    description: 'app.accounts.sidebar.contact: Title for the Contact Us entry in the menu'
+  },
+  profile: {
+    defaultMessage: 'Profile',
+    description: 'app.accounts.sidebar.profile: Title for the Profile entry in the menu'
+  },
+  noAccount: {
+    defaultMessage: 'No Account Loaded',
+    description: 'app.accounts.sidebar.profile: Default account name to show in sidebar when no accounts loaded'
+  },
+})
+
+// To ensure links are safe, we re-use
+// the keys from the routes object.
+type AuthKey = keyof typeof AppRoutes.auth;
+type AuthKeys = {
+  [index in AuthKey]: string
+}
+const authRoutes = Object
+  .keys(AppRoutes.auth)
+  .reduce(
+    (obj, k) => { obj[k as AuthKey] = k; return obj },
+    {} as AuthKeys
+  );
+
+const sidebarLinks: SidebarLink[] = [
+  {
+    name: translations.home,
+    to: `/${authRoutes.home}`,
+    icon: "home"
+  },
+  {
+    name: translations.transferin,
+    to: `/${authRoutes.transferIn}`,
+    icon: "arrow circle up"
+  },
+  {
+    name: translations.makepayments,
+    to: `/${authRoutes.makePayments}`,
+    icon: "arrow circle right"
+  },
+  {
+    name: translations.settings,
+    to: `/${authRoutes.settings}`,
+    icon: "setting"
+  },
+  {
+    name: translations.contact,
+    to: "/contact",
+    icon: "envelope outline"
+  }
+]
+
+SidebarItemsReducer.initialize({
+  items: {
+    header: null,
+    links: sidebarLinks,
+  }
+})
+
+export function useSidebar() {
+
+  const active = AccountMap.useActive();
+  SidebarItemsReducer.useStore();
+  const sidebarApi = SidebarItemsReducer.useApi();
+  const location = useLocation();
+  const intl = useIntl();
+
+
+  // Update the header whenever the active account is changed.
+  React.useEffect(() => {
+    sidebarApi.setHeader({
+      title: translations.profile,
+      avatar: getAvatarLink("14"),
+      accountName: active?.name ?? intl.formatMessage(translations.noAccount),
+      address: active?.address ?? "-----------",
```

Comment:
Is this right? The "-----------"

Reply:
well... I mean... I'm not entirely sure what we could show instead...

The main thing is that the sidebar shouldn't show if we don't have an account loaded.  So it shouldn't ever actually trigger that code.  Maybe in some future world that changes though, so the line should let whoever needs to know that that it's breaking things